packages/deployments/contracts/contracts/nomad-xapps/contracts/promise-router/PromiseMessage.sol

```diff
@@ -118,10 +133,10 @@ library PromiseMessage {
    */
   function isValidPromiseCallbackLength(bytes29 _view) internal pure returns (bool) {
     uint256 _len = _view.len();
-    uint256 _length = lengthOfCalldata(_view);
-    // before = 1 byte identifier + 32 bytes transferId + 20 bytes callback address + 32 bytes length + x bytes data
-    // nonzero callback data
-    return _len > CALLDATA_START && _length > 0 && (CALLDATA_START + _length) == _len;
+    uint256 _length = lengthOfReturndata(_view);
```

Comment:
if you are dealing with an invalid `_view`, then this length could return an incorrect/arbitrary value. this would invalidate the `(RETURNDATA_START + _length) == _len` equality in some edge cases. 

with that being said, it is a bit difficult to create a strong assertion here -- the only thing that would hold is that it is the first statement (`_len > RETURNDATA_START`). maybe we could just add some comments around it about that? curious your thoughts @patitonar @lmcorbalan